### PR TITLE
ROE 2419 Navigate to check your answers page

### PR DESCRIPTION
**Pull request by dlloyd1-ch**
### JIRA link
https://companieshouse.atlassian.net/browse/ROE-2419

### Change description
 Navigate to check your answers page with url params. This is the happy path. Validation and errors to be picked up separately.

### Work checklist

- [x] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/routes/index.ts [hunk 1]
```diff
@@ -340,7 +341,9 @@ router
   });
 
 router.get(config.CHECK_YOUR_ANSWERS_URL, authentication, navigation.hasBOsOrMOs, checkYourAnswers.get);
+router.get(config.CHECK_YOUR_ANSWERS_WITH_PARAMS_URL, authentication, navigation.hasBOsOrMOs, checkYourAnswers.get);
```
Comment (HatiDev) [inline src/routes/index.ts:344]: Changes to router for check your answers page will create conflict with Story for page changes ROE-2386

Comment (dlloyd1-ch) [reply src/routes/index.ts:344]: Agreed. These changes should probably have been down as part of that story, but needed here to enable navigation to 'check your answers'. We can manage the conflict, whichever PR gets merged first.

File: test/controllers/beneficial.owner.type.controller.spec.ts [hunk 2]
```diff
@@ -379,6 +380,28 @@ describe("BENEFICIAL OWNER TYPE controller", () => {
...
+        .post(config.BENEFICIAL_OWNER_TYPE_SUBMIT_URL);
+
+      expect(resp.status).toEqual(302);
+      expect(resp.text).toContain(NEXT_PAGE_URL);
+      expect(resp.header.location).toEqual(NEXT_PAGE_URL);
+      expect(mockGetUrlWithParamsToPath).toHaveBeenCalledTimes(1);
+      expect(mockGetUrlWithParamsToPath.mock.calls[0][0]).toEqual(config.CHECK_YOUR_ANSWERS_WITH_PARAMS_URL);
+    });
+
+    /*
```
Comment (mwestacott) [inline test/controllers/beneficial.owner.type.controller.spec.ts:399]: Is this comment needed here? Or perhaps, rather, can most of the other 'submit' tests be included here too (now). Perhaps not those related to trusts but looks like other scenarios that should work, once this page is delivered.

Actually, some unit tests related to testing and I think @markpit might have spotted something related to validation, so maybe he'll add those tests under a new story?

Comment (markpit) [reply test/controllers/beneficial.owner.type.controller.spec.ts:399]: I can add the validation tests under https://companieshouse.atlassian.net/browse/ROE-2409 but in order to do so I'll have to add/pass the other BO/MO urls into the template besides the current BO Individual.

Comment (dlloyd1-ch) [reply test/controllers/beneficial.owner.type.controller.spec.ts:399]: I left the other 'submit' unit tests out as they all appeared to be validation related and this is just happy path. I can try and get them working here but not sure what value they'll be adding given the validation will be addressed in another ticket.

**Review by markpit — APPROVED** [review]

File: test/controllers/beneficial.owner.type.controller.spec.ts (hunk 2 above)
Comment (mwestacott) [reply test/controllers/beneficial.owner.type.controller.spec.ts:399]: 👍 happy for these to be done by @markpit and that other ticket. Approving.

**Review by mwestacott — APPROVED** [review]

**Review by HatiDev — APPROVED** [review]

**Pull request merged**